matrix/common/etl/__init__.py

```diff
@@ -0,0 +1,219 @@
+        path = os.path.join(output_dir, name)
+        if os.path.isdir(path):
+            for filename in os.listdir(path):
+                _upload_file_to_s3(os.path.join(path, filename), f"{job_id}/{name}/{filename}")
+        else:
+            _upload_file_to_s3(path, f"{job_id}/{name}")
+
+
+def _upload_file_to_s3(path_to_file, s3_prefix):
+    """
+    Uploads a file to S3 preload bucket.
+    :param path_to_file: Local path to file to upload
+    :param s3_prefix: S3 prefix to upload to
+    """
+    try:
+        s3 = boto3.client('s3', region_name='us-east-1')
+        s3.upload_file(path_to_file, os.environ['MATRIX_PRELOAD_BUCKET'], s3_prefix)
+    except Exception as e:
+        print(e)
+
+
+def _populate_all_tables(job_id: str, temp=False):
+    """
+    Creates tables and loads PSVs in S3 into Redshift via SQL COPY.
+    """
+    delete_query_template = """DELETE FROM {0} USING {0}_temp WHERE {0}.{1} = {0}_temp.{1};"""
+    insert_query_template = """INSERT INTO {0} SELECT * FROM {0}_temp;"""
+
+    redshift = RedshiftHandler()
+    transaction = []
+    for table in TableName:
+        if temp and table == TableName.FEATURE:
+            continue
+        s3_prefix = f"s3://{os.environ['MATRIX_PRELOAD_BUCKET']}/{job_id}/{table.value}"
+        iam = os.environ['MATRIX_REDSHIFT_IAM_ROLE_ARN']
+
+        table_name = table.value if not temp else f"{table.value}_temp"
+
+        if table == TableName.FEATURE:
+            copy_stmt = f"COPY {table_name} FROM '{s3_prefix}' iam_role '{iam}' COMPUPDATE ON;"
+        elif table == TableName.CELL:
+            copy_stmt = f"COPY {table_name} FROM '{s3_prefix}' iam_role '{iam}' GZIP COMPUPDATE ON;"
+        elif table == TableName.EXPRESSION or table == TableName.CELL:
+            copy_stmt = f"COPY {table_name} FROM '{s3_prefix}' iam_role '{iam}' GZIP COMPUPDATE ON COMPROWS 10000000;"
+        else:
+            copy_stmt = f"COPY {table_name} FROM '{s3_prefix}' iam_role '{iam}';"
+
+        if temp:
+            logger.info(f"ETL: Building queries to update {table.value} table")
+            transaction.extend([
+                CREATE_QUERY_TEMPLATE[table.value].format("TEMP ", "_temp", table_name),
+                copy_stmt,
+                delete_query_template.format(table.value, RedshiftHandler.PRIMARY_KEY[table]),
+                insert_query_template.format(table.value)
+            ])
+        else:
+            logger.info(f"ETL: Building queries to load {table_name} table")
+            transaction.extend([CREATE_QUERY_TEMPLATE[table.value].format("", table_name),
+                                copy_stmt])
+
+    logger.info(f"ETL: Populating Redshift tables. Committing transaction.")
+    try:
+        redshift.transaction(transaction)
+    except psycopg2.Error as e:
+        # TODO: put this in a DLQ
+        logger.error("Failed to populate Redshift tables. Rolling back.", e)
+
+
+def get_dss_client(deployment_stage: str):
```

Comment:
can we move this to a utils maybe? This same function will also exist in the functional tests

Reply:
Sure, will refactor when it's reintroduced in the tests